codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -832,8 +846,9 @@ public void flowController(F flowController) {
         }
 
         private void checkNewStreamAllowed(int streamId) throws Http2Exception {
-            if (goAwaySent() || goAwayReceived()) {
-                throw connectionError(PROTOCOL_ERROR, "Cannot create a stream since the connection is going away");
+            if (goAwayReceived()) {
```

Comment:
> I suspect that most implementations will want GOAWAY = no new streams. Perhaps we need to make this behavior pluggable somehow?

If an application wants this behavior, it can just simply not create any new streams after it sent a  GOAWAY?